deepinv/optim/data_fidelity.py

```diff
@@ -122,11 +135,13 @@ class StackedPhysicsDataFidelity(DataFidelity):
 
     """
 
-    def __init__(self, data_fidelity_list):
+    def __init__(self, data_fidelity_list: list[DataFidelity]):
```

Comment:
```suggestion
    def __init__(self, data_fidelity_list: list[DataFidelity, ...]):
```
I think it has varying length